backend/BalanceBuddy/src/main/java/com/example/balancebuddy/config/PasswordEncoderConfiguration.java

```diff
@@ -0,0 +1,16 @@
+package com.example.balancebuddy.config;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+@Configuration
+public class PasswordEncoderConfiguration {
+    // BCrypt is used for encoding because it will internally generate a random salt, so this means that each
+    // call will have a different result, therefore the password needs to be encoded only once.

```

Comment:
You can leave in the comments "BCrypt is used for hashing the password" or no comments at all because it is a well known password-hashing technique.